lib/spack/spack/build_environment.py

```diff
@@ -321,12 +321,32 @@ def set_build_environment_variables(pkg, env, dirty=False):
         extra_rpaths = ':'.join(compiler.extra_rpaths)
         env.set('SPACK_COMPILER_EXTRA_RPATHS', extra_rpaths)
 
-    # Add bin directories from dependencies to the PATH for the build.
-    bin_dirs = reversed(filter(os.path.isdir, [
```

Comment:
i am not a python guru, but it looks to me that the original code should have added `bin` dire of `node-js`.